src/storage.rs

```diff
@@ -179,6 +179,29 @@ impl Storage {
         }
     }
 
+    // get scripts hash that should be filtered below the given block number
+    pub fn get_scripts_hash(&self, block_number: BlockNumber) -> Vec<Byte32> {
+        let key_prefix = Key::Meta(FILTER_SCRIPTS_KEY).into_vec();
+        let mode = IteratorMode::From(key_prefix.as_ref(), Direction::Forward);
+
+        self.db
+            .iterator(mode)
+            .take_while(|(key, _value)| key.starts_with(&key_prefix))
+            .filter_map(|(key, value)| {
+                let stored_block_number = BlockNumber::from_be_bytes(
+                    value.as_ref().try_into().expect("stored BlockNumber"),
+                );
+                if block_number >= stored_block_number {
```

Comment:
```suggestion
                if stored_block_number < block_number {
```
`block_number = start_number + limit = prove_state_block_number + 1`, so the largest `stored_block_number` can be `prove_state_block_number + 1`, which is not accurate.